src/cryptonote_protocol/quorumnet.cpp

```diff
@@ -88,6 +89,10 @@ struct QnetState {
     // FIXME:
     //std::chrono::steady_clock::time_point last_blink_cleanup = std::chrono::steady_clock::now();
 
+    std::mutex pulse_message_queue_mutex;
+    std::condition_variable pulse_message_queue_cv;
+    std::queue<pulse::message> pulse_message_queue;
+
```

Comment:
See comments in pulse.cpp: basically I think we can avoid all of this by refactoring pulse.cpp to be event-based.  Then instead of storing anything here you would just immediately queue a job for the tagged pulse thread.